api_v3/lib/types/KalturaCategory.php

```diff
@@ -506,6 +520,26 @@ private function validateCategory(category $sourceObject = null)
 			kuserPeer::createKuserForPartner($partnerId, $this->owner);
 		}
 		
+		if (($this->isAggregationCategory && $this->aggregationCategories) ||
+			($this->isAggregationCategory && $sourceObject && $sourceObject->getAggregationCategories()) ||
+			($this->aggregationCategories && $sourceObject && $sourceObject->getIsAggregationCategory()))
+			{
+				throw new KalturaAPIException(KalturaErrors::AGGREGATION_CATEGORY_WRONG_ASSOCIATION);
+			}
+			
+		if ($this->aggregationCategories)
+		{
+			$aggrCatIdsToCheck = explode (',' , $this->aggregationCategories);
+			foreach ($aggrCatIdsToCheck as $aggrCatIdToCheck)
+			{
+				$agrrCat = categoryPeer::retrieveByPK($aggrCatIdToCheck);
```

Comment:
If category not found the API will return `AGGREGATION_CATEGORY_WRONG_ASSOCIATION`? If so, we should add another validation and return `CATEGORY_NOT_FOUND`